src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -272,42 +272,24 @@ private static string GetArgTypesString(Expression[] arguments)
 
         private static PropertyInfo FindProperty(Type type, string propertyName, Expression[] arguments, BindingFlags flags)
         {
-            var props = type.GetProperties(flags).Where(x => x.Name.Equals(propertyName, StringComparison.CurrentCultureIgnoreCase)); ;
```

Comment:
These two lines were apparently put in place as a substitution for the missing `FindMembers`, see http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/IndexExpression.cs,a2f0d8a96d3c8f44, leading to a bunch of strange logic with redundant checks and `Map` calls. Given that `FindBestProperty` is only used here, I decided to fuse both methods and simplify logic.

`FindMembers` with `Type.FilterNameIgnoreCase` should have the same behavior as our loop with the string equality check using `StringComparison.OrdinalIgnoreCase`, but also saves an array allocation. We don't need LINQ anymore either.